UnityProject/Assets/Scripts/UI/PlayerHealthUI.cs

```diff
@@ -52,27 +66,74 @@ void UpdateMe()
 				}
 				return;
 			}
-
-			if (PlayerManager.LocalPlayerScript.playerHealth.IsRespiratoryArrest && !oxygenAlert.gameObject.activeInHierarchy)
+			hasOxygen = !PlayerManager.LocalPlayerScript.playerHealth.IsRespiratoryArrest && !PlayerManager.LocalPlayerScript.playerHealth.respiratorySystem.IsSuffocating;
+			if (!hasOxygen && !oxygenAlert.gameObject.activeInHierarchy)
 			{
 				oxygenAlert.gameObject.SetActive(true);
 			}
 
-			if (!PlayerManager.LocalPlayerScript.playerHealth.IsRespiratoryArrest && oxygenAlert.gameObject.activeInHierarchy)
+			if (hasOxygen && oxygenAlert.gameObject.activeInHierarchy)
 			{
 				oxygenAlert.gameObject.SetActive(false);
 			}
 		}
 
+		if (monitorTemp && PlayerManager.LocalPlayer != null)
```

Comment:
why do you check if `PlayerManager.LocalPlayer != null` four times? just `return` if it's false or whatever

Reply:
This has been fixed